analyses/molecular-subtyping-MB/02-compare-classes.Rmd

```diff
@@ -0,0 +1,221 @@
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", medullo.classifier.uncorrected.acc))
+
+# output table
+viewDataTable(medullo.classifier.uncorrected)
+```
+
+### Consensus (batch-corrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge mm2s and medulloPackage output
+consensus.corrected <- mm2s.corrected %>%
+  rename(MM2S_best_fit = best.fit) %>%
+  select(-c(classifier, dataset, match, score)) %>%
+  inner_join(medullo.classifier.corrected %>%
+               rename(medulloPackage_best_fit = best.fit) %>%
+               select(-c(classifier, dataset, match, p.value)),
+             by = c('Kids_First_Biospecimen_ID', 'sample_id',
+                    'pathology_subtype','other_molecular_findings'))
+
+# add consensus molecular subtype if both classifiers agree 
+# match consensus subtype to the pathology report
+consensus.corrected  <- consensus.corrected %>%
+  mutate(molecular_subtype = ifelse(MM2S_best_fit == medulloPackage_best_fit, 
+                                    MM2S_best_fit, NA)) %>%
+  mutate(match = str_detect(pathology_subtype, molecular_subtype))
+
+# % accuracy
+consensus.corrected.acc <- consensus.corrected %>%
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", consensus.corrected.acc))
+
+# output table
+viewDataTable(consensus.corrected)
+```
+
+### Consensus (uncorrected)
+
+```{r, echo = TRUE, warning = FALSE, message = FALSE}
+# merge mm2s and medulloPackage output
+consensus.uncorrected <- mm2s.uncorrected %>%
+  rename(MM2S_best_fit = best.fit) %>%
+  select(-c(classifier, dataset, match, score)) %>%
+  inner_join(medullo.classifier.uncorrected %>%
+               rename(medulloPackage_best_fit = best.fit) %>%
+               select(-c(classifier, dataset, match, p.value)),
+             by = c('Kids_First_Biospecimen_ID', 'sample_id',
+                    'pathology_subtype','other_molecular_findings'))
+
+# add consensus molecular subtype if both classifiers agree 
+# match consensus subtype to the pathology report
+consensus.uncorrected  <- consensus.uncorrected %>%
+  mutate(molecular_subtype = ifelse(MM2S_best_fit == medulloPackage_best_fit, 
+                                    MM2S_best_fit, NA)) %>%
+  mutate(match = str_detect(pathology_subtype, molecular_subtype))
+
+# % accuracy
+consensus.uncorrected.acc <- consensus.uncorrected %>%
+  filter(!is.na(pathology_subtype)) %>%
+  group_by(match) %>%
+  summarise(n = n()) %>%
+  mutate(Accuracy = paste0(n/sum(n)*100, '%')) %>%
+  filter(match) %>%
+  .$Accuracy
+print(paste0("Accuracy: ", consensus.uncorrected.acc))
+
+# output table
+viewDataTable(consensus.uncorrected)
+```
```

Comment:
I would expect the output from this notebook to include a table of the consensus labels (preferably as a TSV). 

In modules for subtyping other histologies, we've joined all the identifiers (DNA and RNA) into a single table: https://github.com/AlexsLemonade/OpenPBTA-analysis/blob/e8fbc7dc6aa8a36d7236fbe10a038657e3782e09/analyses/molecular-subtyping-HGG/results/HGG_molecular_subtype.tsv This particular step can come in a subsequent notebook if you'd prefer. But I do not expect this notebook to be overly long, particularly with the custom function changes.

Reply:
Updated markdown to save consensus subtype output merged with RNA+DNA id to tsv files: https://github.com/komalsrathi/OpenPBTA-analysis/blob/mb-class-nb/analyses/molecular-subtyping-MB/02-compare-classes.Rmd#L256
and 
https://github.com/komalsrathi/OpenPBTA-analysis/blob/mb-class-nb/analyses/molecular-subtyping-MB/02-compare-classes.Rmd#L279

Also updated the README at the analyses and module level.